src/main/java/com/github/manolo8/darkbot/gui/drawables/ConstantEntitiesDrawer.java

```diff
@@ -34,22 +34,32 @@ public void onDraw(MapGraphics mg) {
 
     public void drawPortals(MapGraphics mg) {
         mg.setColor("portals");
+        mg.setFont("small");
 
         for (Portal portal : portals) {
             mg.drawOvalCentered(portal, 12.0, false);
+            portal.getTargetMap()
+                    .ifPresent(gameMap -> mg.drawString(portal, gameMap.getShortName(), -8, MapGraphics.StringAlign.MID));
         }
     }
 
     public void drawBattleStations(MapGraphics mg) {
         for (BattleStation bs : battleStations) {
-            if (bs instanceof BattleStation.Asteroid) mg.setColor("meteroid");
-            else if (bs.getEntityInfo().isEnemy()) mg.setColor("enemies");
-            else mg.setColor("allies");
+            if (bs instanceof BattleStation.Asteroid) {
+                mg.setColor("meteroid");
+            } else {
+                mg.setColor(bs.getEntityInfo().isEnemy() ? "enemies" : "allies");
+            }
 
             if (bs instanceof BattleStation.Module) {
                 mg.drawRectCentered(bs, 3.0, false);
             } else {
                 mg.drawOvalCentered(bs, 11.0, 9.0, true);
+
+                String name = bs instanceof BattleStation.Hull ? ("[" + bs.getEntityInfo().getClanTag() + "] ") : "";
+                name += bs.getEntityInfo().getUsername();
+
+                mg.drawString(bs, name, bs instanceof BattleStation.Hull ? -14 : -10, MapGraphics.StringAlign.MID);
```

Comment:
I think this one should go behind a display toggle, or be included in the "show usernames" one, as it can be sensitive info (ie: reveal what server you're playing on in the bot ui)